lesson-1-intro-to-k8s/README.md

```diff
@@ -0,0 +1,243 @@
+Dependencies resolved.
+Nothing to do.
+Complete!
+--> dbdfdd757c0
+STEP 3/4: USER 48
+--> 2c1e273cec3
+STEP 4/4: CMD ["python3", "-m", "http.server", "--bind", "0.0.0.0", "--directory", "/", "9999"]
+COMMIT http-server
+--> bdd326fe09d
+Successfully tagged localhost/http-server:latest
+bdd326fe09d41a497a2b74082aeb6c609b7da7dc7a6b8d749e2d4f53a19e9ff9
+```
+
+And now let's run a container:
+```
+$ podman run http-server
+
+// this is actually blank, no output here
+```
+
+And in a separate console:
+```
+$ curl http://0.0.0.0:9999
+curl: (7) Failed to connect to 0.0.0.0 port 9999 after 0 ms: Connection refused
+```
+
+(you can also try open that URL in your browser)
+
+Also, do you know what "0.0.0.0" and "localhost" mean?
+
+Anyway, what's wrong up there? And what the heck are we actually trying to do?
+
+1. The demonstration shows that your application with all its dependencies,
+   content and metadata is packaged in a container image which is independent
+   of the host OS - similar, but more lightweight, to virtual machines.
+2. The HTTP content is being served within the container and that's not exposed
+   to all network interfaces.
+3. The commands above utilize rootless podman containers and those [don't get
+   their own IP address](https://podman.io/getting-started/network).
+4. [Podman](https://podman.io/) is a container manager (actually a pod manager
+   😁) that doesn't require any service or a daemon to run - it's just a tool.
+4. The resolution is to expose the HTTP server within the container to all the
+   network interfaces.
+
+```
+$ podman run -p 9999:9999 http-server
+```
+
+```
+$ curl http://0.0.0.0:9999
+<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN" "http://www.w3.org/TR/html4/strict.dtd">
+<html>
+<head>
+<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
+<title>Directory listing for /</title>
+</head>
+<body>
+<h1>Directory listing for /</h1>
+<hr>
+<ul>
+<li><a href="bin/">bin@</a></li>
+<li><a href="boot/">boot/</a></li>
+<li><a href="dev/">dev/</a></li>
+<li><a href="etc/">etc/</a></li>
+<li><a href="home/">home/</a></li>
+<li><a href="lib/">lib@</a></li>
+<li><a href="lib64/">lib64@</a></li>
+<li><a href="lost%2Bfound/">lost+found/</a></li>
+<li><a href="media/">media/</a></li>
+<li><a href="mnt/">mnt/</a></li>
+<li><a href="opt/">opt/</a></li>
+<li><a href="proc/">proc/</a></li>
+<li><a href="root/">root/</a></li>
+<li><a href="run/">run/</a></li>
+<li><a href="sbin/">sbin@</a></li>
+<li><a href="srv/">srv/</a></li>
+<li><a href="sys/">sys/</a></li>
+<li><a href="tmp/">tmp/</a></li>
+<li><a href="usr/">usr/</a></li>
+<li><a href="var/">var/</a></li>
+</ul>
+<hr>
+</body>
+</html>
+```
+
+```
+10.0.2.100 - - [22/Apr/2022 08:23:05] "GET / HTTP/1.1" 200 -
+```
+
+
+## Cluster access
+
+Link TBD
+
+Open the URL and [log in](https://docs.openshift.com/container-platform/3.11/dev_guide/authentication.html#dev-guide-authentication) using the credentials we provided.
```

Comment:
Outdated docs!

Reply:
uhhh, thanks!

google keeps pointing me to 3 -_-